### DISCUSSION TITLE
I2S-DMA compatible PCB/Shield board

### DISCUSSION THREAD

**Discussion by vortigont** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#96] [Ideas]
Got sick of those crappy dupont wires hanging around. So I was thinking it would be nice to have a PCB for an easy attaching to the 64x32 or 64x64 modules.
@hallard I saw you have a shield your [repo](https://github.com/hallard/WeMos-Matrix-Shield). Does it works fine with I2S-DMA  lib?
cc @mrfaptastic

**Comment by hallard** [comment]
@vortigont nope, I designed it for ESP8266 (with output connector going back to the panels, don't like either)
Even if we can put ESP32 D1 mini, there is not enough wired GPIO to works with DMA, so I planned to make another one with DMA feature. 
In the meantime I saw [Electrodragon's](https://www.electrodragon.com/product/rgb-matrix-panel-drive-interface-board-for-esp32-dma/) one (not the best but price is unbeatable $3) so I ordered some, even if I don't like the connector (too high) it works flawlessly. Need to use with Pico D4 dev Board or ESP32 dev Board
I had a discussion with Chow from Electrodragon and we talked about features and we merged ideas and they build a new one with LDR, another connector type, and a build in buffers. I should receive samples soon.

I must admit even with no chip buffer on board it works so fine 
![image](https://user-images.githubusercontent.com/2471931/109010797-86a1bb80-76b0-11eb-98ed-b6e28dcf8063.png)

![image](https://user-images.githubusercontent.com/2471931/109010880-a1743000-76b0-11eb-8a5e-05cccc2b1dd3.png)

But if you want me to update mine it's no problem, the deal is if I put buffer on board, users may have trouble to solder these tiny SMD chip (because I want to keep the board tiny). Other way could be a board refresh with wiring for DMA with no buffer chip. Not sure these chip are mandatory with 1 or 2 panels, never had any issues without them. What's your though?

**Reply by vortigont** [reply]
Nice infoboard indeed! :)

> we merged ideas and they build a new one with LDR, another connector type, and a build in buffers. I should receive samples soon
that's great, pls share you experience afterwards. Yeah, current board does not look convenient to use, but the price does! :) Anyway better than dupont jumpers

**Comment by mrcodetastic** [comment]


**Reply by vortigont** [reply]
Cool! With all the efforts combined it makes much better job! :)
I really love that simple approach! You should definitely put those projects/gerbers to the repo along with a lib!
The only changes I could think of is adding some more holes for I2C bus and rx/tx pins

**Reply by hallard** [reply]
ah ah ah that's amazing indeed, and I love the D1 Pro Mini32 version because these ESP32 boards are cheaper than any other one (half price than devkit) and size are really small.
I also like things to be simple and small.

@mrfaptastic interesting you made a board with different mapping than the library you wrote :-)

So what is your advice for wiring in case of rework? keep same as your board or use default lib one ? What GPIO could be considered as "standard" for E line?

**Reply by mrcodetastic** [reply]
Nuking my post as it's easier to just use Brian's panel: http://blough.ie/i2smat/ - It's tried and tested and works with the default pin mapping.

**Reply by vortigont** [reply]
Yep, it looks nice! Except shipping fee from tindie... doubles the price for the shield itself :(
Anyway, it would be nice having an open project and gerbers available for modifications.

**Comment by vortigont** [comment]
Also found [this](https://github.com/rorosaurus/esp32-hub75-driver/blob/master/ASSEMBLY.md) nice DIY boards. Gerber files are available but not the project if any changes required.

**Comment by Lofty73** [comment]
I always use the Boards made by. Brian Lough from tindie. Well made and great support

**Comment by hallard** [comment]
Guys,

took some time, got one, small, I2C, LDR, several way to power, 2 touch pad, and ESP32 in DMA mode of course, should do the job.

![image](https://user-images.githubusercontent.com/2471931/109204875-2f7e1280-77a6-11eb-8344-8bc04f6aed73.png)

![image](https://user-images.githubusercontent.com/2471931/109204658-e8901d00-77a5-11eb-86fb-bd0a0aa3af27.png)

Let me know if you see something strange or need something else.

**Reply by vortigont** [reply]
This little board looks really nice! I like the idea that bottom part could be cut-off inplace in case space is critical. Awesome!
Just a couple of notes - those i2c holes at the top of the board looks a little bit misaligned, is that intentional? and I was thinking about making 1 or 2 pin pads at the top right above power connector. For nor reason, just because there is a free space :)

**Reply by hallard** [reply]
For holes, correct look at this, it perming to fix devices and headers without soldering for test, awesome.
https://www.sparkfun.com/tutorials/114

I'm changing a little bit things adding real connector like this one
![image](https://user-images.githubusercontent.com/2471931/109314347-82f56c80-7849-11eb-92b1-1e8cedda57b9.png)

And on the free space, top right, just putting a diode and jumper to bypass if needed.

Cuffoff bottom part? I didn't tougth about that, shame on me, I made this part for touch pad, but that's a nice idea, I'll add some holes so cutting may be easier :-)

**Reply by vortigont** [reply]
> For holes, correct look at this, it perming to fix devices and headers without soldering for test, awesome.

Hey! That never came my mind before! It's just brilliant! :+1:   :+1:

**Comment by hallard** [comment]
Ok here final version:

- sacrified one touch pad to allow this super usefull big connector
- exposed as asked GPIO32 and GPIO33 top right
- added diode (huge one to allow 5A) to allow avoiding powering panel thru USB with jumper to bypass if needed
- added cutting holes to be able to reduce board if really needed

Schematics
![image](https://user-images.githubusercontent.com/2471931/109335626-98c35b80-7862-11eb-9d4e-28976f7a1e73.png)

And PCB
![image](https://user-images.githubusercontent.com/2471931/109335855-e5a73200-7862-11eb-98a7-8257a3de8c54.png)

I will order a bunch and send you both one if you're interested.

**Reply by vortigont** [reply]
Actually I don't get the idea of that diode neither here nor in Brian's board. If you leave just the jumper without diode than anyway you will be able to power panel via micro-USB (bad idea, but anyway). D1 board already has this diode between micro-USB and +5 V pin, so no harm to the USB source anyway. And having such a large diode is even more useless, you do not need it to power esp32 itself, and if you are going to power panel from USB than anyway you will either short it or be limited with a tiny diode on the D1 board.
So, just curious - what's the use case?
On anything else - I like it! Great job Charles!

**Reply by hallard** [reply]
Well, to be honest powering panels thru USB of D1 is a bad idea (even if I do that everyday) because depending on where do you connect, sometimes power is not enough and resulting of reset or corrupted display. I also tried some USB adapters capable of 2A (such as smart phone chargers) or more with corrupted and wrong display. 
Even with this kind of product (6A they said) with 3000uF added on panel I had issue with just 2 panels consuming (displayed on the 1A) so 5V should be very stable.
![image](https://user-images.githubusercontent.com/2471931/109383709-71af6d00-78e8-11eb-8be0-f8b791ac6a4c.png)

So @mrfaptastic number's one rule, use always good and verified power
So until you know exactly pro and con of powering thru USB it's fine, but for standard user, please don't do that. 

The diode just here to avoid USB 5V going to panels, resulting in `no display`  if not using good 5V power connected to terminal blocks as a kind of reminder power the MATRIX with external power. Of course D1 mini is already protected with on board diode so this diode is not for protecting D1, but more even protect against poor USB computer port, usb hub or poor smart charger with bad over current protection.

When used the diode is for allowing current going to D1 only, and you are correct 5A is oversized, that's what's happening when I get too focus on Matrix consumption, hahaha, my fault I need to resize it with a smaller one of course :-)

**Reply by vortigont** [reply]
Regarding power supply just to share my experience I'm using 12V power supply and a nice DC-DC buck converter right before the panel. To my surprise it works pretty well, I'm powering four 64x64's that way. 12 volts PSUs are much easier to find than beefy 5 volts. And that DCDC is capable of driving 7 amps constantly and about 10 amps in peaks for short periods.
Would love to have some of your PCB's in possesion ) Pls, let me know if it is possible to preorder some :)

**Reply by hallard** [reply]
Nice idea 12V DC/DC could you share the model, worth trying to put it footprint on PCB if possible.

Just created dedicated [repo](https://github.com/hallard/WeMos-Matrix-Shield-DMA) with link to OSHPark to order PCB, but there is also gerber if you need.

**Reply by mrcodetastic** [reply]
Nice one @hallard. Could there/should be space for an isolating cap across teh VCC (5v) and gnd right next to the ESP, perhaps even space for a smaller one across 3v3volt and ground. I've had real troubles with the ESP32's if I have flashing graphics... need at least two smaller caps next to the power input to the esp module.

**Reply by vortigont** [reply]
> Nice idea 12V DC/DC could you share the model, worth trying to put it footprint on PCB if possible.
> Just created dedicated [repo](https://github.com/hallard/WeMos-Matrix-Shield-DMA) with link to OSHPark to order PCB, but there is also gerber if you need.

Sure, here is the [link](https://aliexpress.ru/item/32934802533.html) to DC-DC is that I use. It's quite large to fit to the shield-board but it can easily drive 2-3 panels and even 4 if not stressing it with full while filling.
My next idea is to try it with some QC/PD trigger board and power it via some type-c laptop charger.
![IMG_20210227_200607](https://user-images.githubusercontent.com/25991192/109395276-97fef800-793c-11eb-9c00-b4ba19fcd24c.jpg)

Thanks for the link! Never tried OSHPark service, but will do if if already have your project uploaded there.

**Reply by vortigont** [reply]
BTW, I removed porality-protection diode from DC-DC since it was most heating element on the board. The DC-DC itself is synchronous and does not heats at all up until 5 amps.

**Reply by hallard** [reply]
Yeah, it's too big and already has terminal block, no way to plug it into the shield anyway. Need wiring in all case. I'll try some thanks for sharing.

**Reply by hallard** [reply]
> Nice one @hallard. Could there/should be space for an isolating cap across teh VCC (5v) and gnd right next to the ESP, perhaps even space for a smaller one across 3v3volt and ground. I've had real troubles with the ESP32's if I have flashing graphics... need at least two smaller caps next to the power input to the esp module.


I tried to get some space for a 3300uF but it would requires bigger board, and as we got 3 ways to power the board we can also use one just to plug big capacitor between 5V and GND it could be done at 3 different places.
But most concerning to my eyes is the 3.3V one, I missed that one and correct could improve D1 Mini brownout event. Which values you used? I'm asking to know what type of footprint I need, if it's like 100uF I can get to SMD if bigger I need another footprint such as radial one.

**Reply by hallard** [reply]
I've put some cap's footprint on v1.8a revision, updated [repo](https://github.com/hallard/WeMos-Matrix-Shield-DMA) according to.

**Reply by vortigont** [reply]
The best shield ever! :)

**Reply by vortigont** [reply]
> Yeah, it's too big and already has terminal block, no way to plug it into the shield anyway. Need wiring in all case. I'll try some thanks for sharing.

Tried it today with 24V PSU - it works even better! Efficiency about 87-90%, and it is with some no-name fake capacitors onboard. I believe replaicing it with brand new true LowESR's would make it even better.
I think I will never use 5V PSU's again :)

**Reply by hallard** [reply]
Not sure quite easy to find 24V high power, I searched for ModBus industrial project, 24V DC was not cheap and not easy to find compared to 12V one.
For years I used 12V power chineese marked as good one, but only things I got where issues, so now I pay the price and use good ones.
After tested a lot for Matrix, I'm very happy with my new 5V quality power, very heavy, looks robust, High voltage protection and quality awesome, works fine without any noise (compared to Chineese one)
[MeanWell RS-25-5 25W 5V 5A](https://www.meanwell.com/webapp/product/search.aspx?prod=RS-25) or [MeanWell RS-15-5 15W 5V 3A](https://www.meanwell.com/webapp/product/search.aspx?prod=RS-15) and not expensive. Available quickly here in 2 days with Amazon, just what I wanted.
![image](https://user-images.githubusercontent.com/2471931/109432920-ce587800-7a0d-11eb-91a9-7990f7e52eaa.png)

**Reply by mrcodetastic** [reply]
> Regarding power supply just to share my experience I'm using 12V power supply and a nice DC-DC buck converter right before the panel. To my surprise it works pretty well, I'm powering four 64x64's that way. 

Just to double check. That $1.80 USD 12v to 5v buck converter you [linked](https://aliexpress.ru/item/32934802533.html) to, you're using to power 4 x 64x64 panels!? What is the amperage on the 12volt input size? I have a couple of unused 12volt power supplies from long-dead ADSL routers that I can now put to use if that's the case!

**Reply by vortigont** [reply]
> What is the amperage on the 12volt input size?

Right, that's what I do now. On 12 volts side it is somewhere about 1/2 of 5 volts one. It depends on the picture displayed, about 2-2.5 amps average with spikes up to 3.5-4 amps on effects like plasma with heat palette (the one with lot's of while/yellow colors).
Obviously 1 module is not enough to drive 4 panels at 100% while-fill for a long time. It overheats in less than 1 minute until internal termo protection kicks in. The current soars over 5 amps on 12 volts side in that case.
But for a usual cases like tests and plasma-like effects I do run it 24 hr a day with no issues at all.


> After tested a lot for Matrix, I'm very happy with my new 5V quality power, very heavy, looks robust, High voltage protection and quality awesome

Meanwell's are good ones, agreed. But I mean, with that DC-DC I could use lot's of old/stashed PSU's, including any used laptop's 19.5 volts. Those one's are jewels - usually it is well build branded PSUs in solid a case and could be found at bargain price on a local second-hand scrap yards or sale forums :)

**Reply by hallard** [reply]
Guys, today I received a gift

![image](https://user-images.githubusercontent.com/2471931/111360578-330e0680-868d-11eb-9e5b-63f227d0faa6.png)

![image](https://user-images.githubusercontent.com/2471931/111360631-44efa980-868d-11eb-8d33-8158f93a53ca.png)

But today was my first "throw out PCB" I knew one day it will happen, it was today. I reversed ESP32 side by side connectors so nothing is working, this is why I released new version V1.9. If you already ordered V1.8 boards, I apologize for that.

Check it out
https://github.com/hallard/WeMos-Matrix-Shield-DMA

**Reply by vortigont** [reply]
Sic! Well, no pain - no gain :) It's time to make a new order than. Your boards are the best anyway, Hallard! :) Thanks for sharing

**Reply by mrcodetastic** [reply]
Nice looking PCB @hallard!

**Reply by hallard** [reply]
thanks guy, sorry for the mistake, I ordered 1.9 at Oshpar again and as I ordered other PCBA in china I also ordered 10 units black PCB more, for less than $1/unit great deal, I keep you informed as soon as I got the boards.

I tested Touchpad, works fine except on GPIO2 (the one were it was wired) because GPIO2 is LED of the M&H kit, and also bootstrapping pin, so I changed to another I/O.

**Reply by vortigont** [reply]
Could you pls,  share a link to the factory that you placed an order with a new PCB's @hallard? Or to the project order if possible, so not to upload same gerbers again? I think I will order those new ones at once also :)
Thanks!

**Reply by hallard** [reply]
Sure the ZIP containing gerbers is located here 
https://github.com/hallard/WeMos-Matrix-Shield-DMA/blob/main/gerbers/gerbers.zip

I'm ordering here https://www.cnospcb.com/ but it can be anywhere else.

**Comment by hallard** [comment]
Guys,
Just ti let you know I received V1.9 boards, quickly assembled one

![image](https://user-images.githubusercontent.com/2471931/113333314-8e452780-9322-11eb-9553-9293558fbd73.png)

And, tadaaaaaa, this does work perfectly, LDR is fine, Touch PAD is fine (despite the fact it's under USB connector, so during testing not the easiest way) and I2C is fine also (holes are a bit too high to get full contact without solder)

**Reply by vortigont** [reply]
Awesome! Mines are on it's way :) Awating!
Thanks, @hallard!

**Reply by hallard** [reply]
You're welcome

take care, depending on Matrix type, connector may be too short (for 128x64 for example), in this case use one like that with little spacer.
![image](https://user-images.githubusercontent.com/2471931/113345253-7ecdda80-9332-11eb-9b1b-160db9185355.png)
the small black supplement height is enough

**Reply by vortigont** [reply]
OK, got it! Tnx for the hint! Will check my panel's connector height.

**Comment by bogd** [comment]
Well... since I just received the first working version of my PCBs, here is another option:

[I2S ESP32 D1 Mini PCB](https://github.com/bogd/esp32-morphing-clock/tree/main/pcb)

Features:
* connects to the matrix via a ribbon cable (to avoid issues with the PCB hitting the plastic frame of the matrix). Because I couldn't find those long-legged/spacer connectors anywhere locally :)
* exposes multiple I2C connectors for various sensors

A big **thank you!!** here to @hallard - your board has been the inspiration for this, and looking at the files you shared gave me the courage to attempt to build my own version. 

If you find any glaring errors/mistakes in the design, please let me know :)

**Comment by witnessmenow** [comment]
Hey guys,

I've been working on a new board that will be open source when it's finished, the ESP32 Trinity. It's basically the I2S shield I have but with an ESP32 built in and much better support for being powered from USB.

![image](https://user-images.githubusercontent.com/1562562/123168993-a4b9d580-d470-11eb-9435-ab7aaf069c19.png)
![image](https://user-images.githubusercontent.com/1562562/123169013-aaafb680-d470-11eb-8a52-ca5661fccd90.png)

I'm currently testing what I hope is the final revision, I'll keep you posted!

![image](https://user-images.githubusercontent.com/1562562/123169137-ca46df00-d470-11eb-985c-7226c03e9b18.png)

**Reply by bogd** [reply]
@witnessmenow - While I'm a big fan of your work (and this new board also seems great), I have to say... you seem keen on adding challenges for the newbies (those of us who still solder components by hand, and order PCBs in the smallest quantity possible, so that automated assembly is not really an option :) ). 

With my limited soldering skills, I'm getting chills just looking at that USB connector... 😛

**Reply by mrcodetastic** [reply]
@witnessmenow - Good to have you back. This is looks great, and I like the humour about where it's most likely made haha!

**Reply by witnessmenow** [reply]
Cheers! I have been doing bits and pieces for the last while but I missed a whole raft of awesome changes to the library. I love the new config process, I just need to go back and update any sketches/examples I have now. Also need to update documentation I have.

Ha, thanks. It's a play on the apple "designed in California" thing, I always thought it was kind of strange to have that in big writing and then small writing fro where it was made.

**Comment by witnessmenow** [comment]
Haha, Oh yeah this is for designed for being assembled, my plan is to sell
them fully assembled so people can literally drop them into their matrix

For people interested in doing their own, and not interested in doing 0402
components the majority of the components, including the usb, are available
on jlc assembly service, I tested it a couple of months ago with an earlier
revision, but I don't know what stock levels are like now with the chip
shortage.

I kind of suck at assembling stuff (I'll be getting them made by someone
else) but even I was able to assemble them while on a live stream so it's
doable

(Very long live stream, don't expect people to watch it 😅)
https://youtu.be/0VtTJB6Cxx4
